File: 백준/Gold/31963. 두 배/두 배.py
from math import *
n = int(input())
l = list(map(int, input().split()))
ary = [0 for i in range(n)]
cnt = 0
x = 0
for i in range(1, len(l)):
  ary[i] = max(ceil(log2(l[i - 1] / l[i])) + ary[i - 1], 0)
  cnt += ary[i]
print(cnt)

File: 백준/Gold/31963. 두 배/두 배.cc
#include <stdio.h>
#include <iostream>
#include <algorithm>
#include <math.h>

using namespace std;

int main()
{
	int n, ary[250000], cnt = 0, a;
	cin >> n;
	scanf("%d", &ary[0]);
	for (int i = 1; i < n; i++) {
		scanf("%d", &ary[i]);
		if (ary[i - 1] > ary[i]) {
			a = ceil(log2(ary[i - 1] / double(ary[i])));
			ary[i] *= pow(2, a);
			cnt += a;
		}
	}
	printf("%d", cnt);
}